Core/src/org/sleuthkit/autopsy/ingest/IngestJobExecutor.java

```diff
@@ -1248,100 +1246,185 @@ void addDataArtifacts(List<DataArtifact> artifacts) {
                 }
-            }
+            });
         }
     }
 
     /**
      * Updates the current data source level ingest progress bar with a new task
-     * name, where the task name is the "subtitle" under the display name.
+     * name, where the task name is the "subtitle" under the display name, if
+     * the job has not been cancelled.
      *
      * @param currentTask The task name.
      */
     void advanceDataSourceIngestProgressBar(String currentTask) {
         if (usingNetBeansGUI && !jobCancelled) {
-            synchronized (dataSourceIngestProgressLock) {
+            SwingUtilities.invokeLater(() -> {
                 if (dataSourceIngestProgressBar != null) {
                     dataSourceIngestProgressBar.progress(currentTask);
                 }
-            }
+            });
         }
     }
 
     /**
      * Updates the current data source level ingest progress bar with a new task
-     * name and the number of work units performed, if in the determinate mode.
-     * The task name is the "subtitle" under the display name.
+     * name and the number of work units performed, if in the determinate mode,
+     * and the job has not been cancelled. The task name is the "subtitle" under
+     * the display name.
      *
      * @param currentTask The task name.
      * @param workUnits   Number of work units performed.
      */
     void advanceDataSourceIngestProgressBar(String currentTask, int workUnits) {
         if (usingNetBeansGUI && !jobCancelled) {
-            synchronized (dataSourceIngestProgressLock) {
+            SwingUtilities.invokeLater(() -> {
                 if (dataSourceIngestProgressBar != null) {
                     dataSourceIngestProgressBar.progress(currentTask, workUnits);
                 }
-            }
+            });
+        }
+    }
+
+    /**
+     * Switches the file ingest progress bar to determinate mode, using the
+     * estimated number of files to process as the number of work units.
+     */
+    private void switchFileIngestProgressBarToDeterminate() {
+        if (usingNetBeansGUI) {
+            SwingUtilities.invokeLater(() -> {
+                if (fileIngestProgressBar != null) {
+                    fileIngestProgressBar.switchToDeterminate((int) estimatedFilesToProcess);
+                }
+            });
+        }
+    }
+
+    /**
+     * Updates the current file ingest progress bar upon start of analysis of a
+     * file, if the job has not been cancelled, if the job has not been
+     * cancelled.
+     *
+     * @param fileName The name of the file.
+     */
+    private void updateFileIngestProgressForFileTaskStarted(String fileName) {
+        if (usingNetBeansGUI && !jobCancelled) {
+            SwingUtilities.invokeLater(() -> {
+                if (processedFiles <= estimatedFilesToProcess) {
+                    fileIngestProgressBar.progress(fileName, (int) processedFiles);
```

Comment:
With the synchronization removed, I believe there is a chance that processedFiles will increase between these two lines, leading to the possibility of processedFiles being greater than the maximum for the progress bar.

Reply:
A code experiment (passing MAX_INT) revealed that if processedFiles > estimatedFilesToProcess, then the progress just pegs at 100%, no exception is thrown. This is the same thing that happens in the else clause. So this code can actually be simplified. The simplification is in the next commit.